PyPDF2/_writer.py

```diff
@@ -794,6 +810,16 @@ def write(self, stream: StreamType) -> None:
         self._write_trailer(stream)
         stream.write(b_(f"\nstartxref\n{xref_location}\n%%EOF\n"))  # eof
 
+    def write(self, fileobj):
+
+        if isinstance(fileobj, str) and fileobj:
+            self.fileobj = FileIO(fileobj, "wb")
+            self.my_file = True
```

Comment:
similar to the `__merger.write` function, if the user passes in a string for `fileobj`, we should close it after calling `self.write_stream`, just to be explicit about the handling, rather than rely on python implicitly closing it.